### PR TITLE
Themes: Remove old isPremiumTheme util, use isThemePremium selector instead

### PR DESCRIPTION
**Pull request by ockham**
Removing stuff that has become obsolete as of #8785

### CODE REVIEW

File: client/state/themes/selectors.js [hunk 1]
```diff
@@ -357,7 +357,7 @@ export function getThemeDetailsUrl( state, theme, siteId ) {
  * @return {?String}        Theme setup instructions URL
  */
 export function getThemeSupportUrl( state, theme, siteId ) {
-	if ( ! theme || ! isPremiumTheme( theme ) ) {
+	if ( isJetpackSite( state, siteId ) || ! theme || ! isThemePremium( state, theme.id ) ) {
```
Comment (budzanowski) [inline client/state/themes/selectors.js]: wpcom user lands on Pressable after AT, he has a Premium theme that he has bought while on wpcom. That theme has support url. I think that he should still see that link but now `isJetpackSite( state, siteId )` blocks it.

File: client/state/themes/selectors.js [hunk 2]
```diff
@@ -400,7 +400,7 @@ export function getThemeHelpUrl( state, theme, siteId ) {
  * @return {?String}        Theme purchase URL
  */
 export function getThemePurchaseUrl( state, theme, siteId ) {
-	if ( ! isPremiumTheme( theme ) ) {
+	if ( isJetpackSite( state, siteId ) || ! isThemePremium( state, theme.id ) ) {
```
Comment (budzanowski) [inline client/state/themes/selectors.js]: So you cannot buy premium wpcom theme on jetpack?

**Comment by budzanowski** [comment]
Two questions, otherwise :+1:

File: client/state/themes/selectors.js (hunk 1 above)
Comment (ockham) [inline client/state/themes/selectors.js]: Thinking about it, this is trickier than it seems 😄 

The `getThemeSupportUrl` returns the URL of the theme sheet's 'Setup' tab, which displays the text contained in the theme object's `supportDocumentation` attr, which JP sites don't return. (But as discussed yesterday, we might query WP.com for the theme and merge attributes from it, much like we do for WP.org already.)

Finally, keep in mind that we plan to filter themes that were installed from WP.com from the 'Local Theme' list, so it'll be only in the WP.com themes list below, where I think we'll set the `siteId` prop to `false` to signal we're looking for the generic list (not quite sure about this yet).

For now, I'll try it without the `isJetpackSite` check locally and see if it points me to an empty 'Setup' tab. If it doesn't, I'll remove the check 😄

File: client/state/themes/selectors.js (hunk 2 above)
Comment (ockham) [inline client/state/themes/selectors.js]: Not yet, at least. Not quite sure about how this is going to be handled with Pressable.

File: client/state/themes/selectors.js (hunk 1 above)
Comment (ockham) [inline client/state/themes/selectors.js]: I think I'd like to leave it in for now and revisit once we've settled what to do about that 'Setup' tab.

**Pull request merged**